### PR TITLE
[Tests-only] Refactor "filter by tag" test from fileTest.js 

### PR DESCRIPTION
**Pull request by SwikritiT**
This PR 
- refactors `it('checking method: filter by tag'` so that it only tests `getFilesByTags`  and doesn't rely on  `fileInfo`, ` getFileId` `createTag`, ` tagFile` and the respected interactions
- create new tests for `fileInfo`, `createTag`, ` tagFile`

part of : https://github.com/owncloud/owncloud-sdk/issues/789

### CODE REVIEW

**Review by dpakach — COMMENTED** [review]
lgtm :+1:

File: tests/filesTest.js [hunk 1]
```diff
@@ -1341,6 +1339,53 @@ describe('Main: Currently testing files management,', function () {
...
+              dProp.appendElement('oc:fileid', '', '')
+            }).appendElement('oc:filter-rules', '', ocFilterRules => {
+              ocFilterRules.appendElement('oc:systemtag', '', tagId)
+            })
+          })
+        })
+        .willRespondWith(getFileInfoBy('tag'))
+
+      await getCapabilitiesInteraction(provider)
+      await getCurrentUserInformationInteraction(provider)
```
Comment (saw-jan) [inline tests/filesTest.js:1369]: Can you move these to just after the `createProvider()`
we want to check those interactions before other interactions in a test.
(do similar for other added tests)

**Review by saw-jan — CHANGES_REQUESTED** [review]

**Review by saw-jan — COMMENTED** [review]
LGTM:+1:

**Review by jasson99 — COMMENTED** [review]
LGTM

**Review by saw-jan — APPROVED** [review]

File: tests/filesTest.js [hunk 2]
```diff
@@ -1363,6 +1409,47 @@ describe('Main: Currently testing files management,', function () {
...
+          status: 207,
+          headers: applicationXmlResponseHeaders,
+          body: new XmlBuilder('1.0', '', 'd:multistatus').build(dMultistatus => {
+            dMultistatus.setAttributes({
+              'xmlns:d': 'DAV:',
+              'xmlns:s': 'http://sabredav.org/ns',
+              'xmlns:oc': 'http://owncloud.org/ns'
+            })
+            dMultistatus.appendElement('d:response', '', dResponse => {
+              dResponse.appendElement('d:href', '', `/remote.php/${data === fileId ? 'webdav' : 'dav/files/' + config.adminUsername}/${testFolder}/${testFile}`)
```
Comment (individual-it) [inline tests/filesTest.js:1440]: please make that line shorter

**Review by individual-it — CHANGES_REQUESTED** [review]
looks good, just a small comment

**Review by individual-it — APPROVED** [review]

**Pull request merged**